src/test/java/de/tum/in/www1/artemis/util/AbstractArtemisIntegrationTest.java

```diff
@@ -139,4 +142,14 @@ public void mockGetRepositoryPathFromRepositoryUrl(String projectPath, VcsReposi
     public void mockGetProjectKeyFromAnyUrl(String projectKey) {
         doReturn(projectKey).when(urlService).getProjectKeyFromRepositoryUrl(any());
     }
+
+    /**
+     * Allows to test the number of database queries during a REST call by passing in the REST call and returning a QueryCountAssert object
+     *
+     * @param call the REST call during which the number of database queries will be tracked
+     * @return a QueryCountAssert object allowing to test how many queries were done during the call
+     */
+    protected <T, E extends Exception> QueryCountAssert<T, E> assertThatDb(ThrowingProducer<T, E> call) {
```

Comment:
Could you maybe find a better method names? I think, e.g., `assertQueryCount(() -> {}).equalsAndReturnResult(34)` would be more readable. What do you think?

Reply:
I understand what you mean but I actually think the current names are more clear:

`assertThatDb(request).hasBeenCalledTimes(5);`

You would read this as **assert that the DB has been called 5 times**.